ci/matrix.yaml

```diff
@@ -258,36 +265,76 @@ non_exploded_tags:
 # Jobs that have an implied prerequisite 'build' job:
 build_required_jobs:
   - 'test'
+  - 'test_gpu'
+  - 'test_cpu'
+  - 'test_nolid'
+  - 'test_lid0'
+  - 'test_lid1'
+  - 'test_lid2'
 
 # Jobs that require a GPU
 gpu_required_jobs:
   - 'test'
+  - 'test_gpu'
+  - 'test_nolid'
+  - 'test_lid0'
+  - 'test_lid1'
+  - 'test_lid2'
```

Comment:
suggestion: We should give these more meaningful names. Even when I already know roughly what these mean, I still have no idea what they mean :)

Reply:
The names are friendlier in UI's, etc:

![image](https://github.com/NVIDIA/cccl/assets/58744/d60718b5-b3ff-46fd-b7c4-e6e58e271b18)

It's just the intermediate job names / presets that use the CUB test naming conventions directly.

Reply:
This will be improved by #1758 by including the friendly name with the rest of the job definition.